network/v1alpha1/types_dnsnameresolver.go

```diff
@@ -0,0 +1,108 @@
+}
+
+// DNSNameResolverSpec is a desired state description of DNSNameResolver.
+type DNSNameResolverSpec struct {
+	// name is the DNS name for which the DNS name resolution information will be stored.
+	// For a regular DNS name, only the DNS name resolution information of the regular DNS
+	// name will be stored. For a wildcard DNS name, the DNS name resolution information
+	// of all the DNS names, that matches the wildcard DNS name, will be stored.
+	// For a wildcard DNS name, the '*' will match only one label. Additionally, only a single
+	// '*' can be used at the beginning of the wildcard DNS name. For example, '*.example.com.'
+	// will match 'sub1.example.com.' but won't match 'sub2.sub1.example.com.'
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=^(\*\.)?([A-Za-z0-9-]+\.)*[A-Za-z0-9-]+\.$
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="spec.name is immutable"
+	Name string `json:"name"`
+}
+
+// DNSNameResolverStatus defines the observed status of DNSNameResolver.
+type DNSNameResolverStatus struct {
+	// resolvedNames contains a list of matching DNS names and their corresponding IP addresses
+	// along with TTL and last DNS lookup time.
+	// +listType=map
+	// +listMapKey=dnsName
+	// +patchMergeKey=dnsName
+	// +patchStrategy=merge
+	// +optional
+	ResolvedNames []DNSNameResolverStatusItem `json:"resolvedNames,omitempty" patchStrategy:"merge" patchMergeKey:"dnsName"`
+}
+
+// DNSNameResolverStatusItem describes the details of a resolved DNS name.
+type DNSNameResolverStatusItem struct {
+	// dnsName is the resolved DNS name matching the name field of DNSNameResolverSpec.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=^(\*\.)?([A-Za-z0-9-]+\.)*[A-Za-z0-9-]+\.$
+	DNSName string `json:"dnsName"`
+	// ips contains the list of IP addresses associated with the dnsName.
+	// +kubebuilder:validation:Required
+	// +listType=set
+	IPs []string `json:"ips"`
+	// ttlSeconds is the minimum time-to-live value among all the IP addresses.
+	// +kubebuilder:validation:Required
+	TTLSeconds int32 `json:"ttlSeconds"`
+	// lastLookupTime is the timestamp when the last DNS lookup was completed.
+	// +kubebuilder:validation:Required
+	LastLookupTime *metav1.Time `json:"lastLookupTime"`
+	// resolutionFailures keeps the count of how many times the DNS resolution failed for the
+	// dnsName field. If the DNS resolution succeeds then the field will be set to zero. Upon
+	// every failure, the value of the field will be incremented by one. Upon reaching a threshold
+	// value, the details about the DNS name will be removed.
+	ResolutionFailures int `json:"resolutionFailures,omitempty"`
+	// conditions provide information about the state of the DNS name.
+	//
+	// These are the supported conditions:
+	//
+	//   * Degraded
+	//   - True if the following conditions are met:
+	//     * The last DNS name resolution failed.
+	//   - False if any of those conditions are unsatisfied.
```

Comment:
You probably don't need this much detail here. I'm struggling to parse what this actually means.

Reply:
I have removed the details.